### DISCUSSION TITLE
[v3] Value in input field wire:model doesn't get updated

### DISCUSSION THREAD

**Discussion by Robertbaelde** [livewire/livewire#6181] [7. Bugs (archived)]
### Livewire version

v3.0.0-beta.6

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

Given a component that has a property bound to an input field using wire:model, when the property gets updated by a method in the component. The value in the input field isn't changed. 

For example: 

`test.blade.php`

```
<div>
    {{$testValue}}

    <br>
    <input name=testValue" type="text" wire:model.blur="testValue">
    <br>

    <a href="#" wire:click="setRandomTestValue">Set random testValue</a>
</div>

```

Test.php

```
<?php

namespace App\Livewire;

use Livewire\Component;

class Test extends Component
{
    public string $testValue = '';

    public function setRandomTestValue(): void
    {
        $this->testValue = (string) rand(1, 1000);
    }

    public function render()
    {
        return view('livewire.test');
    }
}
```

What I expect is, that when pressing the "Set random testValue" button, the value displayed and the value in the input field is updated. However, only the displayed value is updated. And the value in the input field remains the old value. 

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@Robertbaelde thanks for reporting! Any chance you could submit a failing test PR? Thanks!

**Reply by Robertbaelde** [reply]
Here you go. Locally this test work for me with php8.1, but fails with php8.2.. 
https://github.com/livewire/livewire/pull/6242

**Comment by markjaquith** [comment]
I attempted to recreate this issue by writing a test, but the test passed. @Robertbaelde more info needed! 

```php
<?php

namespace Livewire\Tests;

use Livewire\Component;
use Livewire\Livewire;

class InputUpdatedByServerTest extends \Tests\BrowserTestCase
{
    /** @test */
    public function input_with_bound_value_updated_by_server_updates_in_browser()
    {
        Livewire::visit(new class extends Component {
            public string $value = 'old';

            public function changeValue(): void
            {
                $this->value = 'new';
            }

            public function render()
            {
                return <<<'HTML'
                    <div>
                        {{$value}}

                        <br>
                        <input dusk="input" type="text" wire:model.blur="value" />
                        <br>

                        <a href="#" wire:click="changeValue" dusk="change">Change input value to "new" on server</a>
                    </div>
                HTML;
            }
        })
            ->waitForLivewire()->click('@change')
            ->assertInputValue('@input', 'new')
        ;
    }
}

```

**Reply by Robertbaelde** [reply]
@markjaquith It seems that the issue arrises only when some value is typed in wire:model first.
This test fails for me with php8.2 (php8.1 succeeds for some weird reason..) 

```
/** @test */
    public function can_update_wire_model_property_from_component()
    {
        Livewire::visit(new class extends Component {
            public string $value = 'old';

            public function changeValue(): void
            {
                $this->value = 'newvalue';
            }

            public function render()
            {
                return <<<'HTML'
                    <div>
                        {{$value}}

                        <br>
                        <input dusk="input" type="text" wire:model.blur="value" />
                        <br>

                        <a href="#" wire:click="changeValue" dusk="change">Change input value to "new" on server</a>
                    </div>
                HTML;
            }
        })
            ->waitForLivewire()
            ->type('@input', 'some random value')
            ->click('@change')
            ->assertInputValue('@input', 'newvalue');
    }
 ```

https://github.com/livewire/livewire/pull/6242

**Comment by Robertbaelde** [comment]
I tried to research some more. When trying to debug in an actual Laravel project. The component from the test case seems to work. Closing this issue for now, as probably I'm doing something wrong somewhere in the test and in my earlier implementations 🤔
